### PR TITLE
Try not to mount app directory only /public/system

### LINKED ISSUE #3261: Avoid running bower twice for testing
https://gitlab.com/publiclab/plots2/commit/2292036555b77f901256387eaa15e436874a538a

I tried the above patch but it somehow broke Jenkins. We should try again.
Bower is configured (thru `./.bowerrc`) to put libs in `./public/lib` but then we also mount the current dir, so we *shadow* the files with the current directory. That's why we've needed to run bower twice, but we will save some time if we manage to solve this! Maybe putting bower libs outside `/app`?

### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?

> What did you expect to see that you didn't?

### Please show us where to look

https://publiclab.org/ ...

### What's your PublicLab.org username?

> This can help us diagnose the issue: 

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by icarito**
Intends to fix #3261 and also possibly #4849 and other issues arising from mountint the app dir as a volume in container.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] PR is descriptively titled 📑
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by icarito** [comment]
This is passing now! I'll test it out in unstable, @jywarren please review if https://github.com/icarito/plots2/blob/not_mount_app_volume/containers/docker-compose-production.yml#L40-L45 sounds like all directories needed at runtime for _writing_.

**Comment by icarito** [comment]
Tried this in _unstable_ and found 404 for `public/system` files, investigating...

**Comment by icarito** [comment]
Issue was with assets precompilation. This is running in unstable now please try it out!

**Comment by gauravano** [comment]
Hi @icarito, please update the status of the issue. Thanks!

File: Dockerfile [hunk 1]
```diff
@@ -38,3 +38,4 @@ WORKDIR /app
 
 RUN yarn install && yarn upgrade
 RUN passenger-config compile-nginx-engine --connect-timeout 60 --idle-timeout 60
+RUN rake assets:precompile && rake tmp:cache:clear
```
Comment (icarito) [inline Dockerfile]: This is the main insight to put assets inside the Docker container. This way we can push to production without running bundler / yarn in production.

**Comment by icarito** [comment]
Hi! I _believe_ this is ready for merging - it does change fundamentally how we treat assets and in particular how we access local files (specific needed directories only are "mounted") - all else will be inside the container.
We should test this out in unstable - to confirm stuff like uploading files works in all instances.
This is a step in the direction of deploying pre-built containers to production instead of building them on the production server. Also it should fix issues with Gemfile.lock and Yarn.lock when deploying, that @jywarren reported.

**Comment by icarito** [comment]
_Except_ travis just failed on it, checking again!

**Comment by icarito** [comment]
Wow it looks like since I made this, we upgraded to a version of rails that _doesn't use_ `initialize_on_precompile=false` which is what I was doing to be able to compile assets without a database.
I need to find a workaround.

**Comment by icarito** [comment]
There is a workaround for this detailed [here](https://iprog.com/posting/2013/07/errors-when-precompiling-assets-in-rails-4-0):

```
passing a dummy database config while precompiling:

bundle exec rake RAILS_ENV=staging DATABASE_URL=mysql2://none:none@127.0.0.1/none assets:precompile
```

I think we need https://github.com/publiclab/plots2/issues/5422 in order to have a workaround for this. So this is stuck for the moment.

**Pull request closed without merging**